xarray/coding/times.py

```diff
@@ -695,7 +695,8 @@ def encode(self, variable, name=None):
     def decode(self, variable, name=None):
         dims, data, attrs, encoding = unpack_for_decoding(variable)
 
-        if "units" in attrs and attrs["units"] in TIME_UNITS:
+        units = attrs.get("units")
+        if isinstance(units, str) and units in TIME_UNITS:
```

Comment:
I'd probably do
```suggestion
        if units is not None and units in TIME_UNITS:
```
or even
```suggestion
        if attrs.get("units") in TIME_UNITS:
```
(assuming `None` is not a valid time unit. Otherwise we can just use a different default.)

Reply:
This will not solve the original issue, because `attrs["units"]` (and `attrs.get("units")`) is a numpy.ndarray and because of that `units in TIME_UNITS` will produce an error. Also, `isinstance(None, str) is False`. The idea is to ensure that units is an str or, at least, if its hashable, but it is simpler to check for str.